src/components/Circles/ReactionsToolbar.tsx

```diff
@@ -0,0 +1,147 @@
+        setLocalReactions((reactions) => [
+          ...reactions,
+          { type, count: 1, userHasReacted: true },
+        ]);
+        useCreateReaction.mutate({ postId: post.id, type });
+        return;
+      }
+
+      // reaction exists locally but current user has not reacted
+      if (!localReactions[reactionIndex].userHasReacted) {
+        setLocalReactions((reactions) => {
+          reactions[reactionIndex].count += 1;
+          reactions[reactionIndex].userHasReacted = true;
+          return reactions;
+        });
+        useCreateReaction.mutate({ postId: post.id, type: type });
+        return;
+      }
+
+      // reaction exists locally and user has already reacted
+      if (localReactions[reactionIndex].userHasReacted) {
+        setLocalReactions((reactions) => {
+          reactions[reactionIndex].count -= 1;
+          reactions[reactionIndex].userHasReacted = false;
+          return reactions;
+        });
+        useUndoReaction.mutate({ userId: data.id, postId: post.id, type });
+        return;
+      }
+    },
+    [data?.id, localReactions, post.id, useCreateReaction, useUndoReaction],
+  );
+
+  return (
+    <View style={styles.reactionToolbarContainer}>
+      {showPicker && (
+        <Portal>
+          <EmojiModal
+            columns={7}
+            onEmojiSelected={(type) => {
+              handleEmojiSelection(type!);
+              setShowPicker((currentVal) => !currentVal);
+            }}
+            onPressOutside={() => {
+              setShowPicker((currentVal) => !currentVal);
+            }}
+          />
+        </Portal>
+      )}
+      <IconButton
+        size={20}
+        icon="emoticon-happy-outline"
+        iconColor={'rgba(0, 0, 0, .2)'}
+        style={styles.addReactionButton}
+        onPress={() => setShowPicker((currentVal) => !currentVal)}
+      />
+      {localReactions.map((reaction, i) => {
+        if (reaction.count) {
+          return (
+            <View key={i} style={styles.reactionContainer}>
+              <Button
+                mode="contained"
+                compact={true}
+                onPress={() => {
+                  handleEmojiSelection(reaction.type);
+                }}
+                style={styles.reactionContainer}
+                contentStyle={styles.reactionContent}
+                labelStyle={styles.reactionLabel}
+              >
+                {reaction.type} {reaction.count}
+              </Button>
+            </View>
+          );
+        }
+      })}
+    </View>
+  );
+};
+
+const defaultStyles = createStyles('ReactionsToolbar', (theme) => ({
+  reactionToolbarContainer: {
+    flex: 1,
+    flexDirection: 'row',
+  },
+  addReactionButton: {
+    height: 20,
+    width: 20,
+  },
+  emojiPickerContainer: {
+    backgroundColor: 'white',
+    height: '80%',
+  },
+  reactionContainer: {
+    paddingRight: theme.spacing.micro,
+  },
+  reactionContent: {
+    height: 30,
+  },
+  reactionLabel: {
+    paddingTop: 4,
```

Comment:
Using fixed padding values as this correctly positions the text within the button and we wouldn't want these to change.